forge-ai/src/main/java/forge/ai/AiController.java

```diff
@@ -1572,6 +1572,8 @@ public boolean apply(final SpellAbility spellAbility) { //don't include removedA
                 return spellAbility instanceof LandAbility || (spellAbility.getHostCard() != null && ComputerUtilCard.isCardRemAIDeck(spellAbility.getHostCard()));
             }
         });
+        //update LivingEndPlayer
+        player.setHasLivingEnd(CardLists.filter(player.getZone(ZoneType.Library).getCards(), c-> "Living End".equalsIgnoreCase(c.getName())).size() > 0);
```

Comment:
use `Iterables.any` for performance so it can terminate earlier instead of filtering each card by name first